google-cloud-channel-v1/AUTHENTICATION.md

```diff
@@ -0,0 +1,169 @@
+environments, the Service Account credentials can be specified by providing the
+path to the
+[JSON keyfile](https://cloud.google.com/iam/docs/managing-service-account-keys)
+for the account (or the JSON itself) in
+[environment variables](#environment-variables). Additionally, Cloud SDK
+credentials can also be discovered automatically, but this is only recommended
+during development.
+
+## Quickstart
+
+1. [Create a service account and credentials](#creating-a-service-account).
+2. Set the [environment variable](#environment-variables).
+
+```sh
+export CHANNEL_CREDENTIALS=path/to/keyfile.json
+```
+
+3. Initialize the client.
+
+```ruby
+require "google/cloud/channel/v1"
+
+client = ::Google::Cloud::Channel::V1::CloudChannelService::Client.new
+```
+
+## Credential Lookup
+
+The google-cloud-channel-v1 library aims to make authentication
+as simple as possible, and provides several mechanisms to configure your system
+without requiring **Service Account Credentials** directly in code.
+
+**Credentials** are discovered in the following order:
+
+1. Specify credentials in method arguments
+2. Specify credentials in configuration
+3. Discover credentials path in environment variables
+4. Discover credentials JSON in environment variables
+5. Discover credentials file in the Cloud SDK's path
+6. Discover GCP credentials
+
+### Google Cloud Platform environments
+
+When running on Google Cloud Platform (GCP), including Google Compute Engine
+(GCE), Google Kubernetes Engine (GKE), Google App Engine (GAE), Google Cloud
+Functions (GCF) and Cloud Run, **Credentials** are discovered automatically.
+Code should be written as if already authenticated.
+
+### Environment Variables
+
+The **Credentials JSON** can be placed in environment variables instead of
+declaring them directly in code. Each service has its own environment variable,
+allowing for different service accounts to be used for different services. (See
+the READMEs for the individual service gems for details.) The path to the
+**Credentials JSON** file can be stored in the environment variable, or the
+**Credentials JSON** itself can be stored for environments such as Docker
+containers where writing files is difficult or not encouraged.
+
+The environment variables that google-cloud-channel-v1
+checks for credentials are configured on the service Credentials class (such as
+{::Google::Cloud::Channel::V1::CloudChannelService::Credentials}):
+
+1. `CHANNEL_CREDENTIALS` - Path to JSON file, or JSON contents
+2. `CHANNEL_KEYFILE` - Path to JSON file, or JSON contents
+3. `GOOGLE_CLOUD_CREDENTIALS` - Path to JSON file, or JSON contents
+4. `GOOGLE_CLOUD_KEYFILE` - Path to JSON file, or JSON contents
+5. `GOOGLE_APPLICATION_CREDENTIALS` - Path to JSON file
+
+```ruby
+require "google/cloud/channel/v1"
+
+ENV["CHANNEL_CREDENTIALS"] = "path/to/keyfile.json"
+
+client = ::Google::Cloud::Channel::V1::CloudChannelService::Client.new
```

Comment:
`Cloud` isn't usually part of the service module name. Should this be changed to `ChannelService` using an override?

Reply:
I'm not keen on changing names unless it's to remove the version. I know it's not the best, but I also don't think it's our place to police the names or second guess them to the point that we might be different from other languages.

Reply:
SGTM.